tests/test_js_signing.py

```diff
@@ -0,0 +1,55 @@
+"""
+Various tests related to validation for automated signing.
+"""
+
+from js_helper import TestCase
+
+
+class TestSearchService(TestCase):
+    """Tests that warnings related to the search service trigger warnings."""
+
+    def test_changes(self):
+        """Tests that setting changes trigger warnings."""
+
+        def test(obj, prop, stuff, warnings):
+            self.setUp()
+            self.run_script("""%s.%s%s;""" % (obj, prop, stuff))
+            self.assert_failed(with_warnings=warnings)
+
+        objs = ("Cc[''].getService(Ci.nsIBrowserSearchService)",
+                "Services.search")
+
+        for obj in objs:
+            warnings = {"signing_severity": "high",
+                        "id": ("testcases_javascript_actions",
+                               "search_service",
+                               "changes")},
+
+            for prop in "currentEngine", "defaultEngine":
+                yield test, obj, prop, " = foo", warnings
```

Comment:
On Tue, May 05, 2015 at 07:24:06AM -0700, Mathieu Agopian wrote:
>What does that do? It looks like some kind of hackery to "generate" tests for unittest/nose? I don't like it, I'm sure you can do the same with a simple call to your `test()` helper above.
>
>To be honest, I'd rather have two separate tests here, and no loop, no helper.

I'm just following the same pattern as the more-recently-added 
tests. I think the idea is that multiple values can be tested 
without one failure short-circuiting the rest.

I don't really have a strong opinion, though. I was just trying 
to be consistent with newer code.